### PR TITLE
Fixed Mesh constructor and setIndex to accept []

### PR DESCRIPTION
**Pull request by assafcw**
1. Mesh is allowed to accept either material or material[]. ---- was fixed in master, removed in merge.
2. setIndex is allowed to accept an array of indices.

Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x ] Test the change in your own code. (Compile and run.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by assafcw** [comment]
It seems the latest fix I merged with is causing problems, so my original fix should also fix the new problems.

`        this.mesh = new Mesh(link, new MeshPhongMaterial({ color: this.color }));
        this.mesh.material.transparent = true;`

causes this -

`    TS2339: Property 'transparent' does not exist on type 'Material | Material[]'.
  Property 'transparent' does not exist on type 'Material[]'.`

File: types/three/three-core.d.ts [hunk 1]
```diff
@@ -836,6 +836,7 @@ export class BufferGeometry extends EventDispatcher {
 
     getIndex(): BufferAttribute;
     setIndex( index: BufferAttribute ): void;
+	setIndex( index: any[] ): void;
```
Comment (efokschaner) [inline types/three/three-core.d.ts]: Indentation is off here

Comment (efokschaner) [inline types/three/three-core.d.ts]: We can do better than `any[]` here. When an array is passed to setIndex, it ultimately gets passed to a TypedArray constructor see:
https://github.com/mrdoob/three.js/blob/master/src/core/BufferGeometry.js#L58
https://github.com/mrdoob/three.js/blob/35a26f178c523514673d992da1aece23c1cfca6e/src/core/BufferAttribute.js#L387

The official signature for those is basically `ArrayLike<number> | ArrayBufferLike`, see:
https://github.com/Microsoft/TypeScript/blob/cd2ea9a12f2bf42fcd4747782b34543a6c36c08c/lib/lib.es6.d.ts#L2035

**Comment by efokschaner** [comment]
Thanks @assafcw for your contribution. I've requested a couple small improvements in the comments.

**Comment by efokschaner** [comment]
I opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/19577 with the suggestions just in case @assafcw can't get to it.

**Comment by assafcw** [comment]
@efkoschaner - you can go ahead with your fix, it's so minor it doesn't matter who does it.

**Pull request closed without merging**